airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -453,7 +453,8 @@ def _labels_to_key(self, labels):
         try:
             return (
                 labels['dag_id'], labels['task_id'],
-                self._label_safe_datestring_to_datetime(labels['execution_date']))
+                self._label_safe_datestring_to_datetime(labels['execution_date']),
+                labels['try_number'])
```

Comment:
Dict *labels* doesn't contain *try_number* key, as it's not set in *WorkerConfiguration* make_pod. That's the reason why pods are not deleted.
https://github.com/apache/incubator-airflow/blob/v1-10-stable/airflow/contrib/kubernetes/worker_configuration.py#L197

Reply:
D'oh.

Would someone (@vardancse? @aliaksandr-d?) be able to fix this, or at least add a test case that covers this?

Reply:
@ashb I'm looking into it

Reply:
@aliaksandr-d @ashb Is there any bug logged for the same so that I can raise the PR for the same?

Reply:
@aliaksandr-d can you please apply the [patch](https://github.com/apache/incubator-airflow/compare/master...vardancse:k8s_pod?expand=1) and confirm the resolution?